source/server/server.cc

```diff
@@ -534,6 +537,10 @@ void InstanceImpl::initialize(const Options& options,
   // cluster_manager_factory_ is available.
   config_.initialize(bootstrap_, *this, *cluster_manager_factory_);
 
+  for (auto&& bootstrap_extension : bootstrap_extensions_) {
+    bootstrap_extension->serverInitialized(serverFactoryContext());
+  }
+
```

Comment:
This is still too early because cluster manager isn't fully initialized at this moment either. move this right before the guard dog initialization?